sdk/servicebus/azure-servicebus/tests/test_sessions.py

```diff
@@ -908,12 +909,16 @@ def test_session_by_session_client_conn_str_receive_handler_peeklock_abandon(sel
                     message = Message("Handler message no. {}".format(i), session_id=session_id)
                     sender.send(message)
 
-            with sb_client.get_queue_session_receiver(servicebus_queue.name, session_id=session_id, prefetch=0) as receiver:
+            with sb_client.get_queue_session_receiver(servicebus_queue.name, session_id=session_id, prefetch=0, idle_timeout=1) as receiver:
                 message = receiver.next()
                 assert message.sequence_number == 1
                 message.abandon()
-                second_message = receiver.next()
-                assert second_message.sequence_number == 1
+                while True: # we can't be sure there won't be a service delay, so we may not get the message back _immediately_, even if in most cases it shows right back up.
+                    second_message = receiver.next()
```

Comment:
I don't understand the "while True" here..

the "seceond_message" isn't received, "raise" is triggered; if "second_message" is received, the program returns.
looks like "while True" has no effect?

Reply:
the while true is because we sent 3 messages, and are not sure when the message we abandoned will be showing up in the stream due to service latencies and round trip times, so I wrote this in a way that I hoped would make the conceptual goal of the test obvious; "as long as, before running out of messages, you get the one you abandoned, you're good."

If this was not obvious, I'm open to any patterns you think would make this goal more clear.  (To your point I could remove the while true and just call serially 3 times or something to that effect, but at that point, why not loop it :P)

Reply:
okay, now I under stand the goal: abandon the message with sequence 1 and receive that message again.

The variable name "second_message" is a bit misleading me. Although I'm still leaning towards our common pattern of "for loop", but it's up to you whether making the change.
```python
for msg in receiver:
    msg.complete()
    if msg.sequence_number == 1:
        return
    

raise Exception("Did not successfully re-receive abandoned message, sequence_number 1 was not observed.")

Reply:
I don't think I had a good reason for not converting to that pattern other than changing as little of the test as needed to make it work; so I find myself agreeing with you.  I'll adjust it.